firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/datagrams.txt

```diff
@@ -0,0 +1,446 @@
+      THEN("The paylaod returned by the output_datagram's paylaod accessor method is equal to the paylaod from the input_buffer body")
+      THEN("The output_datagram's payload accessor method returns a reference to the payload given in its constructor")
+      THEN("The data in the output_datagram's payload buffer is independent of the data in input_buffer")
+
+    WHEN("A body with a non-empty paylaod buffer and a complete 2-byte header where the sequence field of the header does not match sequence given in output_datagram's constructor")
+      THEN("The transform method reports invalid sequence status")
+      THEN("After the transform method is called, the value returned by the output_datagram's seq accessor method is equal to the sequence field of the input_buffer body's header")
+      THEN("The value returned by the output_datagram's length accessor method is equal to the length field of the input_buffer body's header")
+      THEN("The paylaod returned by the output_datagram's paylaod accessor method is equal to the paylaod from the input_buffer body")
+      THEN("The output_datagram's payload accessor method returns a reference to the payload given in its constructor")
+      THEN("The data in the output_datagram's payload buffer is independent of the data in input_buffer")
+
+    WHEN("A body with a complete 2-byte header, a non-empty payload buffer consistent with the length field of the header is given to the receiver")
+      THEN("The transform method reports ok status")
+      THEN("After the transform method is called, the value returned by the output_datagram's seq accessor method is equal to the sequence field of the input_buffer body's header")
+      THEN("The value returned by the output_datagram's length accessor method is equal to the length field of the input_buffer body's header")
+      THEN("The paylaod returned by the output_datagram's paylaod accessor method is equal to the paylaod from the input_buffer body")
+      THEN("The output_datagram's payload accessor method returns a reference to the payload given in its constructor")
+      THEN("The data in the output_datagram's payload buffer is independent of the data in input_buffer")
+
+Scenario: Datagram Receiver correctly reports incrementing, rollover, and resetting of expected sequence number
+
+  GIVEN("A Datagram receiver of buffer size 254 bytes and expected sequence number equal to 1")      
+  // assertions carried out in GIVEN:
+  // The output_datagram's seq accessor method returns 1
+  // The value returned by the output_datagram's length accessor method is 9
+  // The payload buffer returned by the output_datagram's payload accessor method is '0x00 0x07 0x48 0x17 0xb8 0x67 0x1c 0x45 0x28'  
+
+    WHEN("transform is called on two parseable datagram bodies, whose headers have sequence numbers as 1 & 2 respectively")
```

Comment:
The THENs refer to "the input_buffer", I think they should refer to either "the first input_buffer" or "the second input_buffer". I would suggest introducing "input_buffer" in your WHEN, e.g. WHEN transform is called on two parseable datagram bodies as input_buffers, ...